ui/components/app/hold-to-reveal/hold-to-reveal.js

```diff
@@ -0,0 +1,97 @@
+import React, { useContext, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { CircularProgress } from '@material-ui/core';
+import Button from '../../ui/button';
+import Box from '../../ui/box/box';
+import {
+  ALIGN_ITEMS,
+  DISPLAY,
+  JUSTIFY_CONTENT,
+} from '../../../helpers/constants/design-system';
+import { I18nContext } from '../../../contexts/i18n';
+
+export default function HoldToReveal({
+  buttonText,
+  secondsToHold = 1,
+  revealFinished,
+}) {
+  const t = useContext(I18nContext);
+
+  const [progress, setProgress] = useState(0);
+  const [holdButton, setHoldButton] = useState(false);
+
+  useEffect(() => {
+    if (holdButton && progress < 100) {
+      const timer = setInterval(() => {
+        setProgress((prevProgress) =>
+          prevProgress >= 100 ? 0 : prevProgress + 10,
+        );
+      }, 100 * secondsToHold);
```

Comment:
If we are aiming for seconds here, this number should be `1000`.  Otherwise this is good!

Reply:
Yeah, I see what you mean and I understand your suggestion, but there is something that you should take into concern. **CircularProgress** component has a property called **value** which represents the value of the progress indicator for the determinate variant. This value is between **0** and **100** (fulfilled), which are the limits. The logic I used updates the progress every time (after **100 * secondsToHold** milliseconds => example: 100 * 5s = 500ms) for 10, which is in this case 10% and goes up to the maximum of 100 (100% => example: 5000ms or 5s).
Full example:
If we set secondsToHold to 9s which is 9000ms, the progress is updated every 900ms and after 9 seconds the progress is 100 (fulfilled) and everything is finished. In the meantime this is what happened:
- After the first 900ms the progress is 10.
- After the second 900ms the progress is 20.
- After the third 900ms the progress is 30.
- ...
- After the tenth (last) 900ms the progress is 100.